src/widget/wtracktableview.cpp

```diff
@@ -828,20 +831,49 @@ void WTrackTableView::contextMenuEvent(QContextMenuEvent* event) {
     if (modelHasCapabilities(TrackModel::TRACKMODELCAPS_ADDTOCRATE)) {
         m_pCrateMenu->clear();
         CrateSelectResult allCrates(m_pTrackCollection->crates().selectCrates());
+
+        const QList<TrackId> trackIds = getSelectedTrackIds();
+
+        QHash<CrateId, int> currentCrates(m_pTrackCollection->crates().\
```

Comment:
Unfortunately many queries are executed even if I don't open the _Crates_ menu entry. Each right click on selected tracks now has a noticeable delay when you have many crates. The list of crates with their state should only be populated when the corresponding sub-menu entry becomes visible.

Reply:
Ok, I will look into optimizing it and change the name to crates :)